app/database/queries/breeze_buddy/users.py

```
@@ -51,10 +104,593 @@ async def get_user_by_username(username: str) -> Optional[UserInDB]:
+
+async def check_username_exists(username: str) -> bool:
+    """Check if a username already exists.
+
+    Raises exception on database errors for fail-safe behavior.
+    """
+    query = "SELECT 1 FROM users WHERE username = $1"
+    try:
+        async for conn in get_db_connection():
+            row = await conn.fetchrow(query, username)
+            return row is not None
+        return False  # No connection available
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)